portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -97,6 +117,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.setStatus(HttpServletResponse.SC_CREATED);
   }
 
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response) 
+      throws IOException {
+    String idParameter = getParameter(request, ID_KEY, null);
+    if (idParameter == null) {
+      sendRawTextError(
+          response, HttpServletResponse.SC_BAD_REQUEST, "Please supply an ID.");
+      return;
+    }
+
+    Long idToDelete = tryParseLong(idParameter, null);
+    if (idToDelete == null) {
+      sendRawTextError(
+          response, HttpServletResponse.SC_BAD_REQUEST, "ID is an invalid number.");
+      return;
+    }
+
+    Key commentKey = KeyFactory.createKey("Comment", idToDelete);
+    Entity entityToDelete;
+    try {
+      entityToDelete = datastore.get(commentKey);
+    }
+    catch (EntityNotFoundException e) {
+      sendRawTextError(
+          response, 
+          HttpServletResponse.SC_BAD_REQUEST, 
+          "Comment with the specified ID does not exist.");
+      return;
+    }
+    catch (Exception e) {
+      throw e;
+    }
```

Comment:
Are these lines necessary?

Reply:
Yeah no they aren't at all. Not sure what I was thinking here. Will most likely need to add a few throws to the method signature.